addons/controllers/awsebscsi/awsebscsiconfig_controller.go

```diff
@@ -0,0 +1,168 @@
+	awsebsCSIConfig = awsebsCSIConfig.DeepCopy()
+	cluster, err := r.getOwnerCluster(ctx, awsebsCSIConfig)
+	if cluster == nil {
+		return ctrl.Result{}, err // no need to requeue if cluster is not found
+	}
+
+	return r.reconcileAwsEbsCSIConfig(ctx, awsebsCSIConfig, cluster)
+}
+
+func (r *AwsEbsCSIConfigReconciler) reconcileAwsEbsCSIConfig(ctx context.Context,
+	csiCfg *csiv1alpha1.AwsEbsCSIConfig,
+	cluster *clusterapiv1beta1.Cluster) (result ctrl.Result, retErr error) {
+
+	logger := log.FromContext(ctx)
+
+	patchHelper, err := clusterapipatchutil.NewHelper(csiCfg, r.Client)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	defer func() {
+		if retErr != nil {
+			// don't modify AwsEbsCSIConfig if there is an error
+			return
+		}
+
+		if err := patchHelper.Patch(ctx, csiCfg); err != nil {
+			logger.Error(err, "Error patching AwsEbsCSIConfig")
+			retErr = err
+		}
+	}()
+
+	if !csiCfg.GetDeletionTimestamp().IsZero() {
+		return ctrl.Result{}, nil // deleted
+	}
+
+	if result, err = r.reconcileAwsEbsCSIConfigNormal(ctx, csiCfg, cluster); err != nil {
+		logger.Error(err, "Error reconciling AwsEbsCSIConfig")
+		return result, err
+	}
+
+	return result, nil
+}
+
+func (r *AwsEbsCSIConfigReconciler) reconcileAwsEbsCSIConfigNormal(ctx context.Context,
+	csiCfg *csiv1alpha1.AwsEbsCSIConfig,
+	cluster *clusterapiv1beta1.Cluster) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	ownerRef := metav1.OwnerReference{
+		APIVersion: clusterapiv1beta1.GroupVersion.String(),
+		Kind:       cluster.Kind,
+		Name:       cluster.Name,
+		UID:        cluster.UID,
+	}
+
+	if !clusterapiutil.HasOwnerRef(csiCfg.OwnerReferences, ownerRef) {
+		// csiCfg object is patched in defer func in 'reconcileAwsEbsCSIConfig'
+		csiCfg.OwnerReferences = clusterapiutil.EnsureOwnerRef(csiCfg.OwnerReferences, ownerRef)
+	}
+
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      util.GenerateDataValueSecretName(cluster.Name, constants.AwsEbsCSIAddonName),
+			Namespace: csiCfg.Namespace},
+		Type: v1.SecretTypeOpaque,
+	}
+
+	mutateFn := func() error {
+		secret.StringData = make(map[string]string)
+		dvs, err := r.mapAwsEbsCSIConfigToDataValues(ctx, csiCfg, cluster)
+		if err != nil {
+			logger.Error(err, "Error while mapping AwsEbsCSIConfig to data values")
+			return err
+		}
+		yamlBytes, err := yaml.Marshal(dvs)
+		if err != nil {
+			logger.Error(err, "Error marshaling CSI config data values to yaml")
+			return err
+		}
+		secret.StringData[constants.TKGDataValueFileName] = string(yamlBytes)
+		return nil
+	}
+
+	secret.SetOwnerReferences([]metav1.OwnerReference{ownerRef})
+
+	_, err := controllerutil.CreateOrPatch(ctx, r.Client, secret, mutateFn)
+	if err != nil {
+		logger.Error(err, "Error creating or patching AwsEbsCSIConfig data values secret")
+		return ctrl.Result{}, err
+	}
+
+	csiCfg.Status.SecretRef = &secret.Name
```

Comment:
Why is this a pointer? By doing this, in the status the secret name is not shown, but the secret name's address, right?

Reply:
it should be good to use secret.Name, let me align with other addons